workflow/validate/validate.go

```diff
@@ -225,9 +225,27 @@ func ValidateWorkflow(wftmplGetter templateresolution.WorkflowTemplateNamespaced
 			return err
 		}
 	}
+
+	// Validate OnExit hooks
+	// If the OnExit is specified in Workflow:
+	//   - If the template that referred by OnExit is from a WorkflowTemplate, template in WorkflowTemplate will be validated.
+	//   - If the template is inlined, the template will be validated.
+	// If the OnExit is empty in Workflow:
+	//   - If OnExit is specified in referred WorkflowTemplate, the OnExit of the referred WorkflowTemplate will be validated.
+	//   - If OnExit is empty in referred WorkflowTemplate, nothing will be validated.
+	var tmplHolder *wfv1.WorkflowStep
 	if wf.Spec.OnExit != "" {
+		if hasWorkflowTemplateRef {
+			tmplHolder = &wfv1.WorkflowStep{TemplateRef: wf.Spec.WorkflowTemplateRef.ToTemplateRef(wf.Spec.OnExit)}
+		} else {
+			tmplHolder = &wfv1.WorkflowStep{Template: wf.Spec.OnExit}
+		}
```

Comment:
Would it make sens to set a default and skip the else?
```suggestion
		tmplHolder = &wfv1.WorkflowStep{Template: wf.Spec.OnExit}
		if hasWorkflowTemplateRef {
			tmplHolder = &wfv1.WorkflowStep{TemplateRef: wf.Spec.WorkflowTemplateRef.ToTemplateRef(wf.Spec.OnExit)}
		}
```

Reply:
Done. Thank you!